src/create_state.rs

```diff
@@ -14,7 +15,7 @@ mod config {
 }
 use self::config::{Config, ResultExt};
 
-pub fn create_state() -> Result<State, Error> {
+pub fn create_state() -> Result<(State, SocketAddr), Error> {
```

Comment:
Why not include the bind addr in the state like before?

Reply:
The new `SocketAddr` has to be consumed (systemd socket can only be used once) while the `State` has to be shared. I could introduce a new struct to deal with that but for two values tuple seems good enough.